gapis/replay/end_of_replay2.go

```diff
@@ -0,0 +1,96 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replay
+
+import (
+	"context"
+
+	"github.com/google/gapid/core/data/binary"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/replay/builder"
+	"github.com/google/gapid/gapis/replay/value"
+)
+
+// EndOfReplay2 is a transform that causes a post back at the end of the replay. It can be used to
+// ensure that GAPIS waits for a replay to finish, if there are no postbacks, or no pending
+// postbacks in the replay.
+type EndOfReplay2 struct {
+	res []Result
+}
+
+func NewEndOfReplay2() *EndOfReplay2 {
+	return &EndOfReplay2{
+		res: []Result{},
+	}
+}
+
+// AddResult adds the given replay result listener to this transform.
+func (endTransform *EndOfReplay2) AddResult(r Result) {
+	endTransform.res = append(endTransform.res, r)
+}
+
+func (t *EndOfReplay2) RequiresAccurateState() bool {
+	return false
+}
+
+func (endTransform *EndOfReplay2) BeginTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return inputCommands, nil
+}
+
+func (endTransform *EndOfReplay2) ClearTransformResources(ctx context.Context) {
+	// Do nothing
+}
+
+func (endTransform *EndOfReplay2) EndTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return append(inputCommands, endTransform.CreateNotifyInstruction(ctx, defaultNotifyFunction)), nil
+}
+
+func (endTransform *EndOfReplay2) TransformCommand(ctx context.Context, id api.CmdID, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return inputCommands, nil
+}
+
+func defaultNotifyFunction() interface{} {
+	return nil
+}
+
+// CreateNotifyInstruction creates a command that adds an instruction to the replay stream
+// that will notify GAPIS that the replay has finished.
+// It should be the end (or very near the end) of the replay stream and thus
+// be called from an EndTransform.
+func (endTransform *EndOfReplay2) CreateNotifyInstruction(ctx context.Context, result func() interface{}) api.Cmd {
+	return Custom{T: 0, F: func(ctx context.Context, s *api.GlobalState, b *builder.Builder) error {
+		// Since the PostBack function is called before the replay target has actually arrived at the post command,
+		// we need to actually write some data here. r.Uint32() is what actually waits for the replay target to have
+		// posted the data in question. If we did not do this, we would shut-down the replay as soon as the second-to-last
+		// Post had occurred, which may not be anywhere near the end of the stream.
+		code := uint32(0x6060fa51)
```

Comment:
`0x6060fa51` looks like a "magic value" that should be declared as a constant, with a comment explaining the value, e.g.:

```
// eosCode is a magic int used to check the end-of-stream 
const eosCode = uint32(0x6060fa51)
```

Reply:
rename to eosCode. Make it const if possible.